components/de_amp/browser/de_amp_service.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_DE_AMP_BROWSER_DE_AMP_SERVICE_H_
+#define BRAVE_COMPONENTS_DE_AMP_BROWSER_DE_AMP_SERVICE_H_
+
+#include <memory>
+#include <string>
+
+#include "components/keyed_service/core/keyed_service.h"
+#include "url/gurl.h"
+
+class PrefRegistrySimple;
+class PrefService;
+
+namespace de_amp {
+
+class DeAmpService : public KeyedService {
```

Comment:
Frankly speaking, I don't understand why we need a new KeyedService (+factory to it).
It doesn't have a effective state.
All non-static methods can simply be replaced to a few free functions:
```
  void ToggleDeAmp(PrefService* prefs, bool on);
  void DisableDeAmpForTest(PrefService* prefs);
  bool IsEnabled(PrefService* prefs);
```

Reply:
I was following the pattern in Speedreader and Debounce and other places, I understood that we should use KeyedService when the setting can vary by profile. Will take another look.

Reply:
Yeah I think this is a pretty common pattern followed in BraveContentBrowserClient::CreateURLLoaderThrottles and other places

Reply:
I agree that this class serves no useful purpose